redisson/src/main/java/org/redisson/cluster/ClusterConnectionManager.java

```diff
@@ -658,14 +658,21 @@ private void checkSlotsChange(Collection<ClusterPartition> newPartitions) {
     }
     
     private void checkSlotsMigration(Collection<ClusterPartition> newPartitions) {
+        // https://github.com/redisson/redisson/issues/3635
+        int slotSize = slot2entry.length();
+        MasterSlaveEntry[] slot2EntryCopy = new MasterSlaveEntry[slotSize];
```

Comment:
create copy only if required.

Reply:
Hi, i change the policy. It will copy one entry per partition. 
It's too complex to copy only if required. I think we should keep it simple.

Reply:
And i think we should'nt get MasterSlaveEntry of partition from slot2entry, in multithreading, the MasterSlaveEntry we got maybe override by other partition? the entry we expected could be other's.  Is there a way put MasterSlaveEntry of partition into ClusterPartition?

Reply:
> the MasterSlaveEntry we got maybe override by other partition

topology update executed by single thread.

>  Is there a way put MasterSlaveEntry of partition into ClusterPartition?

I'm afraid no